Tests/UpstreamIntegrationTests/UpstreamIntegrationTests.swift

```diff
@@ -71,50 +67,361 @@ class UpstreamIntegrationTests: TestBase {
+        // Only validate for the location hint relevant to Edge Network extension
+        let expectedLocationHintJSON = #"""
+        {
+          "payload": [
+            {
+              "ttlSeconds" : 1800,
+              "scope" : "EdgeNetwork",
+              "hint" : "stringType"
+            }
+          ]
+        }
+        """#
+
+        assertEdgeResponseHandle(expectedHandleType: TestConstants.EventSource.LOCATION_HINT_RESULT, expectedHandle: expectedLocationHintJSON, exactMatchPaths: ["payload[*].scope"])
+
+        let expectedStateStore1stJSON = #"""
+        {
+          "payload": [
+            {
+              "maxAge": 1,
+              "key": "stringType",
+              "value": "stringType"
+            },
+            {
+              "maxAge": 1,
+              "key": "stringType",
+              "value": "stringType"
+            }
+          ]
+        }
+        """#
+
+        assertEdgeResponseHandle(expectedHandleType: TestConstants.EventSource.STATE_STORE, expectedHandle: expectedStateStore1stJSON)
+    }
+
+    // MARK: - Configuration tests
+    // Tests standard sendEvent with preset location hint
+    func testSendEvent_withSetLocationHint_receivesExpectedEventHandles() {
+        // Setup
+        let locationHint = "va6"
+        let locationHintNetworkRequest = NetworkRequest(urlString: "https://obumobile5.data.adobedc.net/ee/\(locationHint)/v1/interact", httpMethod: .post)!
+        networkService.setExpectationForNetworkRequest(networkRequest: locationHintNetworkRequest, expectedCount: 1)
+
+        Edge.setLocationHint(locationHint)
+
+        let eventPayloadJSON = #"""
+        {
+          "xdm": {
+            "testString": "xdm"
+          },
+          "data": {
+            "testDataString": "stringValue"
+          }
+        }
+        """#
+
+        let xdm = getAnyCodableAndPayload(eventPayloadJSON, type: .xdm)!
+        let data = getAnyCodableAndPayload(eventPayloadJSON, type: .data)!
+
+        let experienceEvent = ExperienceEvent(xdm: xdm.payload, data: data.payload)
+
+        // Test
+        Edge.sendEvent(experienceEvent: experienceEvent)
+
+        // Verify
+        // MARK: Network response assertions
+        let matchedResponse = networkService.getResponsesFor(networkRequest: locationHintNetworkRequest, timeout: 5)
+        XCTAssertEqual(200, matchedResponse.first?.responseCode)
+
+        // MARK: Response Event assertions
+        // Only validate for the location hint relevant to Edge Network extension
+        let expectedLocationHintJSON = #"""
+        {
+          "payload": [
+            {
+              "ttlSeconds" : 1800,
+              "scope" : "EdgeNetwork",
+              "hint" : "\#(locationHint)"
+            }
+          ]
+        }
+        """#
+
+        assertEdgeResponseHandle(expectedHandleType: TestConstants.EventSource.LOCATION_HINT_RESULT, expectedHandle: expectedLocationHintJSON, exactMatchPaths: ["payload[*].scope", "payload[*].hint"])
```

Comment:
is this assert waiting without setting an expectation for this event?

Reply:
in order to optimize the time you wait for one/multiple events we should set the expectations beforehand. example for testSendEvent_withStandardExperienceEventTwice_receivesExpectedEventHandles:
```
expectEdgeEventHandles(handleType: TestConstants.EventSource.LOCATION_HINT_RESULT, expectedCount: 2)
expectEdgeEventHandles(handleType: TestConstants.EventSource.STATE_STORE, expectedCount: 2)

Edge.sendEvent(...)
Edge.sendEvent(...)

let locHandles = getEdgeEventHandles(handleType: TestConstants.EventSource.LOCATION_HINT_RESULT)
XCTAssertEqual(2, locHandles.count)
// assert content for each
```